gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/Path.java

```diff
@@ -238,10 +239,53 @@ public default boolean popEquals(final Pop pop, final Object other) {
                 isPresent();
     }
 
+    /**
+     * Isolate a sub-path from the path object. The isolation is based solely on the path labels.
+     * The to-label is inclusive. Thus, from "b" to "c" would isolate the example path as follows {@code a,[b,c],d}.
+     *
+     * @param fromLabel The label to start recording the sub-path from.
+     * @param toLabel   The label to end recording the sub-path to.
+     * @return the isolated sub-path.
+     */
+    public default Path subPath(final String fromLabel, final String toLabel) {
+        if (null == fromLabel && null == toLabel)
+            return this;
+        else {
+            Path subPath = MutablePath.make();
+            int size = this.size();
```

Comment:
Missed a `final` on `size` - that variable doesn't appear to be re-assigned elsewhere.